packages/xchain-aggregator/src/aggregator.ts

```diff
@@ -86,17 +86,44 @@ export class Aggregator {
     return optimalSwap
   }
 
+  /**
+   * Should be approved
+   * @param errors
+   * Check if the tx should be approved
+   * @returns boolean
+   */
+  private shouldBeApproved(errors: string[]) {
+    return (
+      errors.some((error) => error === 'Thorchain router has not been approved to spend this amount') &&
+      errors.some((error) => error !== 'Asset should be ERC20')
+    )
+  }
+
   /**
    * Do swap
    * @param {QuoteSwapParams & { protocol?: Protocol }} params Swap parameters. If protocol is not set,
    * estimateSwap will be call and swap will be done in protocol with the greatest expected amount
+   * Aprove the tx if needed
```

Comment:
Typo: `approve`

Reply:
Yeah, thank you
[fixed here](https://github.com/xchainjs/xchainjs-lib/pull/1243/commits/d69a67eebda2ed8312422f08bad6b65bbc7a4a10)